pkg/api/helper/helpers.go

```diff
@@ -113,9 +113,18 @@ func IsStandardContainerResourceName(str string) bool {
 }
 
 // IsOpaqueIntResourceName returns true if the resource name has the opaque
-// integer resource prefix.
+// integer resource prefix, or it is another unknown resource name not in the
+// default namespace.
 func IsOpaqueIntResourceName(name api.ResourceName) bool {
-	return strings.HasPrefix(string(name), api.ResourceOpaqueIntPrefix)
+	return IsCustomNamespacedResourceName(name)
```

Comment:
We could alternatively do something like:

```go
func IsIntResourceName(name api.ResourceName) bool {
    return IsOpaqueIntResourceName(name) || IsExtendedResourceName(name)
}
```

And then update current uses of `IsOpaqueIntResourceName()` in the scheduler to use `IsIntResourceName()` instead. This would preserve the existing OIR prefix behavior and resources with the new `"extensions.kubernetes.io/"` prefix ride along with the existing scheduler logic for OIRs.